components/bootloaders/STM/stm32f103/include/HW_CAN.h

```diff
@@ -0,0 +1,318 @@
+                                        CAN_IER_LECIE | CAN_IER_ERRIE)
+
+
+typedef enum CAN_exitCode
+{
+    HAL_CAN_ERROR_NONE            = (0x00000000UL),  // No error
+    HAL_CAN_ERROR_EWG             = (0x00000001UL),  // Protocol Error Warning
+    HAL_CAN_ERROR_EPV             = (0x00000002UL),  // Error Passive
+    HAL_CAN_ERROR_BOF             = (0x00000004UL),  // Bus-off error
+    HAL_CAN_ERROR_STF             = (0x00000008UL),  // Stuff error
+    HAL_CAN_ERROR_FOR             = (0x00000010UL),  // Form error
+    HAL_CAN_ERROR_ACK             = (0x00000020UL),  // Acknowledgment error
+    HAL_CAN_ERROR_BR              = (0x00000040UL),  // Bit recessive error
+    HAL_CAN_ERROR_BD              = (0x00000080UL),  // Bit dominant error
+    HAL_CAN_ERROR_CRC             = (0x00000100UL),  // CRC error
+    HAL_CAN_ERROR_RX_FOV0         = (0x00000200UL),  // Rx FIFO0 overrun error
+    HAL_CAN_ERROR_RX_FOV1         = (0x00000400UL),  // Rx FIFO1 overrun error
+    HAL_CAN_ERROR_TX_ALST0        = (0x00000800UL),  // TxMailbox 0 transmit failure due to arbitration lost
+    HAL_CAN_ERROR_TX_TERR0        = (0x00001000UL),  // TxMailbox 0 transmit failure due to transmit error
+    HAL_CAN_ERROR_TX_ALST1        = (0x00002000UL),  // TxMailbox 1 transmit failure due to arbitration lost
+    HAL_CAN_ERROR_TX_TERR1        = (0x00004000UL),  // TxMailbox 1 transmit failure due to transmit error
+    HAL_CAN_ERROR_TX_ALST2        = (0x00008000UL),  // TxMailbox 2 transmit failure due to arbitration lost
+    HAL_CAN_ERROR_TX_TERR2        = (0x00010000UL),  // TxMailbox 2 transmit failure due to transmit error
+    HAL_CAN_ERROR_TIMEOUT         = (0x00020000UL),  // Timeout error
+    HAL_CAN_ERROR_NOT_INITIALIZED = (0x00040000UL),  // Peripheral not initialized
+    HAL_CAN_ERROR_NOT_READY       = (0x00080000UL),  // Peripheral not ready
+    HAL_CAN_ERROR_NOT_STARTED     = (0x00100000UL),  // Peripheral not started
+    HAL_CAN_ERROR_PARAM           = (0x00200000UL),  // Parameter error
+    HAL_CAN_ERROR_INTERNAL        = (0x00800000UL),  // Internal error
+} CAN_exitCode;
+
+
+typedef enum
+{
+    CAN_TX_MAILBOX_0 = 0U,
+    CAN_TX_MAILBOX_1,
+    CAN_TX_MAILBOX_2,
+    CAN_TX_MAILBOX_COUNT,
+} CAN_TxMailbox_E;
+
+typedef enum
+{
+    CAN_IDENTIFIER_STD = 0U, // Standard length CAN ID
+    CAN_IDENTIFIER_EXT,      // Extended length CAN ID
+} CAN_IdentifierLen_E;
+
+typedef enum
+{
+    CAN_REMOTE_TRANSMISSION_REQUEST_DATA = 0U,
+    CAN_REMOTE_TRANSMISSION_REQUEST_REMOTE,
+} CAN_RemoteTransmission_E;
+
+
+typedef union
+{
+    uint64_t u64;
+    uint32_t u32[2];
+    uint16_t u16[4];
+    uint8_t  u8[8];
+} CAN_data_T;
+
+typedef struct
+{
+    uint16_t                 id;
+
+    CAN_IdentifierLen_E      IDE;
+    CAN_RemoteTransmission_E RTR;
+
+    uint8_t                  lengthBytes;
+    CAN_data_T               data;
+    CAN_TxMailbox_E          mailbox;
+} CAN_TxMessage_t;
```

Comment:
Keep consistent capitalization *_T

Reply:
I think we should stick to lowercase t, like with all the standard types (i.e `uint8_t`)